contracts/src/mocks/MockERC20SendAndCallReceiver.sol

```diff
@@ -0,0 +1,32 @@
+// (c) 2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+// SPDX-License-Identifier: Ecosystem
+
+pragma solidity 0.8.18;
+
+import {IERC20SendAndCallReceiver} from "../interfaces/IERC20SendAndCallReceiver.sol";
+import {SafeERC20TransferFrom} from "@teleporter/SafeERC20TransferFrom.sol";
+import {SafeERC20} from "@openzeppelin/contracts@4.8.1/token/ERC20/utils/SafeERC20.sol";
+import {IERC20} from "@openzeppelin/contracts@4.8.1/token/ERC20/ERC20.sol";
+
+/**
+ * THIS IS AN EXAMPLE CONTRACT THAT USES UN-AUDITED CODE.
+ * DO NOT USE THIS CODE IN PRODUCTION.
+ */
+
+/**
+ * @notice TODO
+ */
+contract MockERC20SendAndCallReceiver is IERC20SendAndCallReceiver {
+    using SafeERC20 for IERC20;
+
+    /**
+     * @dev See {IERC20SendAndCallReceiver-receiveTokens}
+     */
+    function receiveTokens(address token, uint256 amount, bytes calldata payload) external {
+        require(payload.length > 0, "MockERC20SendAndCallReceiver: empty payload");
```

Comment:
(Optional) Would it help to add a "tokenReceived" event to the mock contracts that emits the payload so that the E2E test could check for the event occurrence with the expected data?